packages/lib/src/FormControllerClass/FormControllerClass.ts

```diff
@@ -522,24 +522,52 @@ export class FormControllerClass {
     await this.validate();
 
     const [errors, values] = toJSCompat([this.API.errors, this.API.values]);
-    const isSuccess = isEmpty(errors);
+    const isValid = isEmpty(errors);
+
+    let isSuccess: boolean = false;
+    let error: unknown;
+    let response: unknown;
 
     try {
       if (this.options.onSubmit) {
-        await this.options.onSubmit({errors, values, isSuccess, event: submitEvent});
+        response = await this.options.onSubmit({
+          errors,
+          values,
+          isValid,
+          isSuccess: isValid,
+          event: submitEvent,
+        });
       }
 
+      isSuccess = true;
+
       runInAction(() => {
         if (isEmpty(errors)) {
           this.updateInitialValues();
           this.updateIsDirtyBasedOnFields();
         }
         this.setIsSubmitting(false);
       });
-    } catch {
+    } catch (e) {
+      error = e;
+
       this.setIsSubmitting(false);
     }
 
-    return {errors, values, isSuccess};
+    return {
+      errors,
+      values,
+      isValid,
+      isSuccess: isValid,
+      submitResult: isSuccess
+        ? ({
+            isSuccess,
```

Comment:
keep one isSuccess in function response - in the main object or in submitResult.